### PR TITLE
[WIP] Add ARKV6X

### PR DESCRIPTION
**Pull request by AlexKlimaj**
This PR adds the ARKV6X PAB form factor flight controller. 

One thing we should think about is moving all the PAB carrier configs to a common place. For detecting revision and starting sensors.

https://arkelectron.com/product/arkv6x/

https://review.px4.io/plot_app?log=d85da83d-4dcb-4490-8e6b-57d5e81edac7
https://review.px4.io/plot_app?log=9ed11e11-24f5-4e11-8f9f-6485423a0201
https://review.px4.io/plot_app?log=71e0d26f-a656-4b29-97f5-eedab70537d2

Xp9i8CtCTjakj7GU5trL_IMG_3800
rU27nbgvQLGVOuwWshKy_IMG_3799

### CODE REVIEW

**Comment by mrpollo** [comment]
Hey @davids5 would the PAB suggestion from @AlexKlimaj work?

**Comment by davids5** [comment]
@AlexKlimaj - would you elaborate on you suggestion?

File: boards/ark/fmu-v6x/firmware.prototype [hunk 1]
```diff
@@ -0,0 +1,13 @@
+{
+    "board_id": 57,
```
Comment (davids5) [inline boards/ark/fmu-v6x/firmware.prototype:2]: @AlexKlimaj did you do a reservation PR for this ID in the bootloader repo https://github.com/PX4/PX4-Bootloader/blob/main/board_types.txt#L13?

File: boards/ark/fmu-v6x/nuttx-config/include/board.h [hunk 2]
```diff
@@ -0,0 +1,558 @@
+/************************************************************************************
+ * nuttx-configs/px4_fmu-v6x/include/board.h
```
Comment (davids5) [inline boards/ark/fmu-v6x/nuttx-config/include/board.h:2]: Fix path and Copyright dates

File: boards/ark/fmu-v6x/nuttx-config/include/board.h [hunk 3]
```diff
@@ -0,0 +1,558 @@
...
+#if defined(CONFIG_STM32H7_SDMMC_XDMA) || defined(CONFIG_STM32H7_SDMMC_IDMA)
+#  define STM32_SDMMC_SDXFR_CLKDIV    (5 << STM32_SDMMC_CLKCR_CLKDIV_SHIFT)
+#else
+#  define STM32_SDMMC_SDXFR_CLKDIV    (100 << STM32_SDMMC_CLKCR_CLKDIV_SHIFT)
+#endif
+
+#define STM32_SDMMC_CLKCR_EDGE      STM32_SDMMC_CLKCR_NEGEDGE
+
+/* LED definitions ******************************************************************/
+/* The PX4 FMUV6X board has three, LED_GREEN a Green LED, LED_BLUE a Blue LED and
```
Comment (davids5) [inline boards/ark/fmu-v6x/nuttx-config/include/board.h:305]: Fix Name

**Review by davids5 — CHANGES_REQUESTED** [review]
LGTM other then a few Minor nits.

File: boards/ark/fmu-v6x/nuttx-config/bootloader/defconfig [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+CONFIG_BOARD_RESET_ON_ASSERT=2
+CONFIG_C99_BOOL8=y
+CONFIG_CDCACM=y
+CONFIG_CDCACM_IFLOWCONTROL=y
+CONFIG_CDCACM_PRODUCTID=0x0039
+CONFIG_CDCACM_PRODUCTSTR="ARK BL FMU v6X.x"
+CONFIG_CDCACM_RXBUFSIZE=600
+CONFIG_CDCACM_TXBUFSIZE=12000
+CONFIG_CDCACM_VENDORID=0x3185
+CONFIG_CDCACM_VENDORSTR="ARK"
```
Comment (davids5) [inline boards/ark/fmu-v6x/nuttx-config/bootloader/defconfig:38]: Add BL and fix RE for it
"ARK BL FMU v6X.x"

**Review by davids5 — CHANGES_REQUESTED** [review]
On more

File: boards/ark/fmu-v6x/firmware.prototype (hunk 1 above)
Comment (AlexKlimaj) [reply boards/ark/fmu-v6x/firmware.prototype:2]: https://github.com/PX4/PX4-Bootloader/pull/220

**Comment by AlexKlimaj** [comment]
> @AlexKlimaj - would you elaborate on your suggestion?

At the moment in main, the PAB configurations are duplicated in the V5X and V6X. Once I add this, we have them in three locations to maintain. In the ideal world, we'd do something with the eeprom on the carriers. In the meantime, we should break out the PAB carrier config to a common location, like rc.carrier, where any PAB flight controller can reference. That way we don't need to have all these combinations of SOM hardware rev + carrier rev that we currently have.

File: boards/ark/fmu-v6x/nuttx-config/bootloader/defconfig (hunk 4 above)
Comment (AlexKlimaj) [reply boards/ark/fmu-v6x/nuttx-config/bootloader/defconfig:38]: Wouldn't it be ARK for the VENDORSTR and "BL FMU v6X.x" for the PRODUCTSTR?

File: boards/ark/fmu-v6x/src/spi.cpp [hunk 5]
```diff
@@ -1,6 +1,6 @@
 /****************************************************************************
  *
- *   Copyright (C) 2020, 2022 PX4 Development Team. All rights reserved.
+ *   Copyright (C) 2022, 2022 PX4 Development Team. All rights reserved.
```
Comment (davids5) [inline boards/ark/fmu-v6x/src/spi.cpp:3]: Copyright (C) 2022 PX4 Development Team. All rights reserved.

**Review by davids5 — CHANGES_REQUESTED** [review]

File: boards/ark/fmu-v6x/src/spi.cpp [hunk 6]
```diff
@@ -0,0 +1,86 @@
+/****************************************************************************
+ *
+ *   Copyright (C) 2022, 2022 PX4 Development Team. All rights reserved.
```
Comment (dagar) [inline boards/ark/fmu-v6x/src/spi.cpp:3]:
```suggestion
 *   Copyright (C) 2022 PX4 Development Team. All rights reserved.
```

**Comment by dagar** [comment]
This is mostly self contained, we might as well merge at this point and continue in main to avoid any conflicts.

**Review by dagar — APPROVED** [review]

**Pull request merged**